internal/controllers/machineset/machineset_preflight.go

```diff
@@ -216,6 +228,14 @@ func (r *Reconciler) kubeadmVersionPreflightCheck(cpSemver, msSemver semver.Vers
 	return nil, nil
 }
 
+func (r *Reconciler) controlPlaneVersionPreflightCheck(cpVersion, msVersion string) preflightCheckErrorMessage {
+	if cpVersion != msVersion {
```

Comment:
I'm wondering if we should apply this check only for cluster with a managed topology, because in this case we know that the machine is going to be replaced soon.

Instead, if the cluster in not managed, it is up to the user to execute upgrades, and they can also decide to have a cluster with machine pools of mixed versions for a long time.

Reply:
What if they implemented something simlar to the Cluster topology controller? How would they run this check then if we  hard-code here that we skip it?

Reply:
If we're fine with not allowing folks to run this for regular non-classy Clusters I'm okay with adding another if

Reply:
Nevermind. You're right. We have to add this. Will do